qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -40,14 +41,19 @@
 class IBMQFactory:
     """Factory and credentials manager for IBM Q Experience."""
 
-    def __init__(self):
+    def __init__(self) -> None:
         self._credentials = None
         self._providers = OrderedDict()
         self._v1_provider = IBMQProvider()
 
     # Account management functions.
 
-    def enable_account(self, token, url=QX_AUTH_URL, **kwargs):
+    def enable_account(
+            self,
+            token: str,
+            url: str = QX_AUTH_URL,
+            **kwargs: Dict
+    ) -> Optional[Union[AccountProvider, IBMQProvider]]:
```

Comment:
Once we fully get rid of v1 api this will no longer return `IBMQProvider`. Can you add a note in #268 to remind us to change this type hint once that happens?